### PR TITLE
Add penalty button

### LINKED ISSUE #343: Add penalties handling

### PR DESCRIPTION
**Pull request by amalieeh**
This fixes #343 by adding penalty button.

### CODE REVIEW

**Comment by amalieeh** [comment]
Before: screen shot 2018-02-26 at 19 53 23

After: 
screen shot 2018-02-26 at 19 54 02

screen shot 2018-02-26 at 19 53 50

**Comment by amalieeh** [comment]
screen shot 2018-03-05 at 20 30 49

File: app/routes/users/components/PenaltyForm.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                  component={TextInput.Field}
+                  placeholder={placeholder || 'Hvor tungt vektlegges prikken?'}
+                />
+                <Field
+                  name="sourceEvent"
+                  placeholder="Arrangementer"
+                  filter={['events.event']}
+                  component={SelectInput.AutocompleteField}
+                />
+                <Button submit>SEND</Button>
```
Comment (finnss) [inline app/routes/users/components/PenaltyForm.js]: Should probably be called "Lagre" throughout. For us it's intuitive that clicking this "sends" something from a front-end to a back-end, for the end users probably have no idea about this infrastructure. In that context, "send" sounds like you're sending a message or something, while what they're trying to do is save.

File: app/actions/UserActions.js [hunk 2]
```diff
@@ -393,6 +394,38 @@ export function sendForgotPasswordEmail({
...
+
+export function deletePenalty(id: number) {
+  return callAPI({
+    types: Penalty.DELETE,
+    endpoint: `/penalties/${id}`,
+    method: 'DELETE',
+    schema: penaltySchema,
+    meta: {
+      penaltyId: id,
+      errorMessage: 'Slette prikk feilet'
```
Comment (finnss) [inline app/actions/UserActions.js]: "Sletting av"?

File: app/routes/users/components/Penalties.js [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 import React from 'react';
 import { FormatTime } from 'app/components/Time';
+//import Button from 'app/components/Button';
```
Comment (finnss) [inline app/routes/users/components/Penalties.js]: Remove this if you don't need it

File: app/routes/users/UserProfileRoute.js [hunk 4]
```diff
@@ -40,12 +42,15 @@ const mapStateToProps = (state, props) => {
     // feedItems = selectFeedActivitesByFeedId(state, {
     //   feedId: feedIdByUserId(user.id)
     // });
+
+    penalties = selectPenaltyByUserId(state, { userId: user.id });
```
Comment (SmithPeder) [inline app/routes/users/UserProfileRoute.js]: This could be initialized right away?

File: app/routes/users/components/PenaltyForm.js [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  button?: boolean,
+  className?: string
+};
+
+type State = {
+  hidden: boolean,
+  sent: boolean
+};
+
+class AnnouncementInLine extends Component<Props, State> {
```
Comment (odinuge) [inline app/routes/users/components/PenaltyForm.js]: Maybe rename this? :smile:

File: app/routes/users/components/PenaltyForm.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  props: Props;
+  state = {
+    hidden: true,
+    sent: false
+  };
+
+  onSubmit = (penalty, user) => {
+    //?
```
Comment (odinuge) [inline app/routes/users/components/PenaltyForm.js]: Remove this comment

File: app/routes/users/components/PenaltyForm.js [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+                onSubmit={handleSubmit(values => this.onSubmit(values, user))}
+              >
+                <Field
+                  name="reason"
+                  component={TextArea.Field}
+                  placeholder={
+                    placeholder || 'Skriv din begrunnelse for prikk...'
+                  }
+                />
+                <Field //Nytt
```
Comment (odinuge) [inline app/routes/users/components/PenaltyForm.js]: Remove this comment

File: app/routes/users/components/PenaltyForm.js [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+    this.setState(prevState => ({
+      hidden: !prevState.hidden
+    }));
+  };
+
+  render() {
+    const {
+      actionGrant,
+      handleSubmit,
+      placeholder,
```
Comment (odinuge) [inline app/routes/users/components/PenaltyForm.js]: You can remove this `placeholder` functionality.

Comment (amalieeh) [reply app/routes/users/components/PenaltyForm.js]: Placeholder is used in the different fields in the form, should I replace them with another function?

**Pull request merged**